tests/foreman/destructive/test_katello_agent.py

```diff
@@ -147,3 +148,49 @@ def test_positive_install_and_remove_package_group(katello_agent_client):
     sat.cli.Host.package_group_remove(hammer_args)
     for package in constants.FAKE_0_CUSTOM_PACKAGE_GROUP:
         assert client.run(f'rpm -q {package}').status != 0
+
+
+def test_positive_upgrade_warning(sat_with_katello_agent):
+    """Ensure a warning is dispayed when upgrading with katello-agent enabled.
+
+    :id: 2bfc5e3e-e147-4e7a-a25c-85be22ef6921
+
+    :expectedresults: Upgrade check fails and warning with proper message is displayed.
+
+    :CaseLevel: System
+    """
+    sat = sat_with_katello_agent
+    ver = sat.version.split('.')
+    target_ver = f'{ver[0]}.{int(ver[1]) + 1}'
+    warning = (
+        "The katello-agent feature is enabled on this system. As of Satellite 6.15, katello-agent "
+        "is removed and will no longer function. Before proceeding with the upgrade, you should "
+        "ensure that you have deployed and configured an alternative tool for remote package "
+        "management and patching for content hosts, such as Remote Execution (REX) with pull-based "
+        "transport. See the Managing Hosts guide in the Satellite documentation for more info. "
+        "Disable katello-agent with the command `satellite-installer --foreman-proxy-content-enable"
+        "-katello-agent false` before proceeding with the upgrade. Alternatively, you may skip "
+        "this check and proceed by running satellite-maintain again with the `--whitelist` option, "
+        "which will automatically uninstall katello-agent."
+    )
+
+    upstream_url = 'https://yum.theforeman.org/releases/nightly/el8/x86_64/'
+    upstream_files = get_repo_files_by_url(upstream_url)
+    fm_rpm = [item for item in upstream_files if 'foreman_maintain' in item]
+    assert fm_rpm, 'no upstream foreman-maintain package found'
+
+    for rpm in fm_rpm:
+        res = sat.execute(f'yum -y install {upstream_url}{rpm}')
+        assert res.status == 0, f'{rpm} installation failed'
+
+    sat.execute('satellite-maintain upgrade list-versions')
+    res = sat.execute('satellite-maintain upgrade list-versions')
+    assert res.status == 0, 'list-versions command failed'
+    assert target_ver in res.stdout, 'target version unavailable'
+
+    res = sat.execute(
+        f'satellite-maintain upgrade check --target-version="{target_ver}" '
+        '--whitelist="repositories-setup,repositories-validate" --assumeyes'
+    )
+    assert res.status, 'upgrade check passed unexpectedly'
+    assert warning in res.stdout, 'warning message missing or changed'
```

Comment:
IIRC, we get RC =78 for f-m commands when there's any WARNING in upgrade check or run commands
```suggestion
    assert result.status == 78, 'upgrade check passed unexpectedly'
    assert warning in result.stdout, 'Katello-agent warning message missing or changed'
```

Reply:
NACK, there is no WARNING, it just FAILs directly and returns `1`. Actually, I don't care much if it's 1 or 2 or 78, it just must not pass with `0`.

Reply:
Oh right, this is mentioned as warning, but instead its error message, which fails upgrade check if KA is enabled, so would suggest confirming the RC == 1 instead of non-zero RC check